snap/local/helper-go/configure.go

```diff
@@ -20,90 +20,93 @@ package main
 
-func main() {
+// configure is called by the main function
+func configure() {
 	var debug = false
 	var err error
 	var envJSON string
 
 	status, err := cli.Config("debug")
 	if err != nil {
-		fmt.Println(fmt.Sprintf("edgex-device-rest:configure: can't read value of 'debug': %v", err))
-		os.Exit(1)
+		log.Fatalf("edgex-device-rest:configure: can't read value of 'debug': %v", err)
 	}
 	if status == "true" {
 		debug = true
 	}
 
 	if err = hooks.Init(debug, "edgex-device-rest"); err != nil {
-		fmt.Println(fmt.Sprintf("edgex-device-rest:configure: initialization failure: %v", err))
-		os.Exit(1)
-
+		log.Fatalf("edgex-device-rest:configure: initialization failure: %v", err)
 	}
 
 	envJSON, err = cli.Config(hooks.EnvConfig)
 	if err != nil {
-		hooks.Error(fmt.Sprintf("Reading config 'env' failed: %v", err))
-		os.Exit(1)
+		log.Fatalf("Reading config 'env' failed: %v", err)
 	}
 
 	if envJSON != "" {
 		hooks.Debug(fmt.Sprintf("edgex-device-rest:configure: envJSON: %s", envJSON))
-		err = hooks.HandleEdgeXConfig("device-rest", envJSON, local.ConfToEnv)
+		err = hooks.HandleEdgeXConfig("device-rest", envJSON, ConfToEnv)
 		if err != nil {
-			hooks.Error(fmt.Sprintf("HandleEdgeXConfig failed: %v", err))
-			os.Exit(1)
+			log.Fatalf("HandleEdgeXConfig failed: %v", err)
 		}
 	}
 
 	// If autostart is not explicitly set, default to "no"
 	// as only example service configuration and profiles
 	// are provided by default.
-	autostart, err := cli.Config(hooks.AutostartConfig)
+	autostart, err := snapctl.Get("autostart").Run()
 	if err != nil {
-		hooks.Error(fmt.Sprintf("Reading config 'autostart' failed: %v", err))
-		os.Exit(1)
+		log.Fatalf("Reading config 'autostart' failed: %v", err)
 	}
 	if autostart == "" {
-		hooks.Debug("edgex-device-rest autostart is NOT set, initializing to 'no'")
+		log.Debug("autostart is NOT set, initializing to 'no'")
 		autostart = "no"
 	}
 	autostart = strings.ToLower(autostart)
+	log.Debugf("autostart=%s", autostart)
 
-	hooks.Debug(fmt.Sprintf("edgex-device-rest autostart is %s", autostart))
-
-	// service is stopped/disabled by default in the install hook
+	// services are stopped/disabled by default in the install hook
 	switch autostart {
-	case "true":
-		fallthrough
-	case "yes":
-		err = cli.Start("device-rest", true)
+	case "true", "yes":
+		err = snapctl.Start("device-rest").Enable().Run()
 		if err != nil {
-			hooks.Error(fmt.Sprintf("Can't start service - %v", err))
-			os.Exit(1)
+			log.Fatalf("Can't start service: %s", err)
 		}
-	case "false":
-		// no action necessary
-	case "no":
+	case "false", "no":
 		// no action necessary
 	default:
-		hooks.Error(fmt.Sprintf("Invalid value for 'autostart' : %s", autostart))
-		os.Exit(1)
+		log.Fatalf("Invalid value for 'autostart': %s", autostart)
 	}
 
 	log.SetComponentName("configure")
-	if err := options.ProcessAppConfig("device-rest"); err != nil {
-		hooks.Error(fmt.Sprintf("could not process options: %v", err))
-		os.Exit(1)
+
```

Comment:
`log.SetComponentName("configure")` should be added to the beginning of the function so that all calls to log function use that in the syslog tag.